SonarQube.TeamBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -0,0 +1,191 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="RoslynAnalyzerProvider.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using SonarQube.Common;
+using SonarQube.TeamBuild.Integration;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Xml;
+
+namespace SonarQube.TeamBuild.PreProcessor.Roslyn
+{
+    public class RoslynAnalyzerProvider
+    {
+        public const string RoslynCSharpFormatName = "roslyn-config-cs";
+        public const string RoslynCSharpRulesetFileName = "SonarQubeRoslyn-cs.ruleset";
+
+        public const string CSharpLanguage = "cs";
+        public const string CSharpPluginKey = "csharp";
+        public const string CSharpRepositoryKey = "csharp";
+
+        private readonly ISonarQubeServer server;
+        private readonly TeamBuildSettings settings;
+        private readonly string projectKey;
+        private readonly ILogger logger;
+
+        #region Public methods
+
+        /// Sets up the client to run the Roslyn analyzers as part of the build
+        /// i.e. creates the Roslyn ruleset and provisions the analyzer assemblies
+        /// and rule parameter files
+        /// </summary>
+        public static CompilerAnalyzerConfig SetupAnalyzers(ISonarQubeServer server, TeamBuildSettings settings, string projectKey, ILogger logger)
```

Comment:
The SonarLintAnalyzerProvider created one well-known ruleset and unpacked one well-known analyser, so it wasn't necessary to return this information - the targets file could recalculate it. The configuration is more complex now so we'll return the values explicitly.